tensorflow/python/keras/layers/wrappers_test.py

```diff
@@ -290,6 +291,43 @@ def call(self, inputs):
     td3 = keras.layers.TimeDistributed(NoReshapeLayer())
     self.assertFalse(td3._always_use_reshape)
 
+  @tf_test_util.run_in_graph_and_eager_modes
+  def test_TimeDistributed_output_shape_return_types(self):
+
+    class TupleOutputShapeLayer(keras.layers.Layer):
+
+      def call(self, inputs):
+        return inputs
+
+      def compute_output_shape(self, input_shape):
+          return (input_shape[0], input_shape[1])
+
+    class ListOutputShape(keras.layers.Layer):
+
+      def call(self, inputs):
+        return inputs
+
+      def compute_output_shape(self, input_shape):
+          return [input_shape[0], input_shape[1]]
+
+    class TensorShapeOutputShapeLayer(keras.layers.Layer):
+
+      def call(self, inputs):
+        return inputs
+
+      def compute_output_shape(self, input_shape):
+          return tensor_shape.TensorShape([input_shape[0], input_shape[1]])
+
+    test_layers = [TupleOutputShapeLayer, ListOutputShapeLayer,
+                   TensorShapeOutputShapeLayer]
+    # Custom layers can specify output shape as a list, tuple, or TensorShape
+    for layer in test_layers:
+      td = layer()
+      ph = keras.backend.placeholder(shape=(None, 1, 13))
+      out = td(ph)
+      # all output shapes are exposed in the end as TensorShape instances
+      self.assertEqual(isinstance(out, tensor_shape.TensorShape), True)
```

Comment:
Since the out here is an output tensor, I don't feel the assertion here is correct. Shouldn't u use the time distributed wrapper to wrap those test layer and assert the output shape? Checking the type here is probably not enough.

Also if you want check the type, probably should use compute_output_shape explicitly.

Reply:
I admit the test is speculative based on the surrounding tests. I haven't been able to run it locally yet because my local bazel build failed. I will execute the test and verify that it conforms to your suggested changes.

Reply:
This is still not correct, the output is a time distributed wrapped input, which is not a tensor shape.
You should assert the shape of the output, instead of the type of the output.

Reply:
Thanks for the review, and sorry for the confusion; I pushed the first commit eagerly before I read all of your comments. I've now incorporated your feedback and verified the test passes locally with Bazel.